openedx/edx-ora2
# Jkantor/flexible override

**TL;DR -** Make ORA peer grading respect the course-level flexible override setting

JIRA: [AU-1288](https://2u-internal.atlassian.net/browse/AU-1288)

**What changed?**

- Created a method in the workflow mixin,  get_course_workflow_settings, which returns a dictionary, that currently only has a single key, force_on_flexible_peer_openassessments. It is intended to be a place where any future course-level fields that affect ORAs to be read from.
- Pass the value from that through .... too much stuff. We need to pass that data through several levels of API calls to get it to where it's actually needed. There is no way currently to make this less awkward.
- When we check if flex grading is turned on, we now check if the course flag is true, and then check the block.

**Developer Checklist**


**Testing Instructions**

- Turn off the flex-grading course-wide flag
- Make an ORA that requires you to grade 1 person and be graded by 2 people, and is not flexibly graded.
- Submit submissions as two learners. They should each grade each other. They should not get grades.
- Turn on the flex-grading course-wide flag
- Manually edit one of the submissions to have been submitted more than 7 days ago (submission.submitted_at)
- Log in as each user. The learner whose submission was thrust back into the past should now have a grade. The learner who did not, should still not have a grade.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
